src/main/java/build/buildfarm/instance/stub/StubInstance.java

```diff
@@ -240,23 +251,90 @@ public void write(byte[] b, int off, int len) throws IOException {
 
   @Override
   public InputStream newStreamInput(String name) {
-    throw new UnsupportedOperationException();
+    Iterator<ReadResponse> replies = bsBlockingStub
+      .get()
+      .read(ReadRequest.newBuilder().setResourceName(name).build());
+    return new InputStream() {
+      InputStream chunkIn = nullInputStream;
+      boolean closed = false;
+
+      @Override
+      public int available() throws IOException {
+        if (closed) {
+          throw new IOException("stream is closed");
+        }
+        if (chunkIn.available() == 0) {
+          advance();
+        }
+        return chunkIn.available();
+      }
+
+      @Override
+      public int read() throws IOException {
+        if (closed) {
+          throw new IOException("stream is closed");
+        }
+        byte[] b = new byte[1];
```

Comment:
return super.read()?

Reply:
```
src/main/java/build/buildfarm/instance/stub/StubInstance.java:263: error: abstract method read() in InputStream cannot be accessed directly
        return super.read();
```

Gotta have my simple implementation, it seems